include/mqt-core/dd/Operations.hpp

```diff
@@ -186,17 +141,17 @@ getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
 // An empty permutation marks the identity permutation.
 
 template <class Config>
-qc::MatrixDD getDD(const qc::Operation* op, Package<Config>& dd,
+qc::MatrixDD getDD(const qc::Operation& op, Package<Config>& dd,
```

Comment:
Just a comment: I was a bit hesitant about this change at first, as I thought it would cause object slicing, but I convinced myself that this is not the case and this still works as intended. I agree with your assessment that this even makes the intent clearer because `op == nullptr` was never even handled in the first place.